### ISSUE TITLE
Mobile - Position of Text Input and Text objects are not consistent

### ISSUE THREAD

**Issue by broknecho** [Scirra/Construct-3-bugs#5042]
## Problem description

When positioning Text Input and Text objects overtop of Sprite, their position doesn't stay consistent.  Html Controls overtop of Canvas don't seem that stable.

## Attach a .c3p

[C3MobilePositionError.c3p.zip](https://github.com/Scirra/Construct-3-bugs/files/7108739/C3MobilePositionError.c3p.zip)

## Steps to reproduce

1. Open Project
2. Export Project as HTML5
3. Host project somewhere you can open it on as many different devices as wanted
4. Observe that the Text Input and Text label do not sit in the same size/position on it's background Sprite as per what you see in the editor.

I have deployed it here if you want to use that for a test:  https://www.digitalwilson.com/c3mobileerror/

## Observed result

Here's a Blisk attempt to emulate a bunch of iOS devices that I don't own:

![image](https://user-images.githubusercontent.com/3253547/132070928-ca6d19e8-a34f-4b95-9874-f2f9a85184d4.png)

In practice on my project, this has been quite a bit worse then what that image shows:
![image](https://user-images.githubusercontent.com/3253547/132071010-8ae332ef-200c-434f-852a-6d1b1271c479.png)

That "OK" is lined up in the center on Android devices but NOT in iOS devices.

![image](https://user-images.githubusercontent.com/3253547/132071048-e840a93e-d785-41ad-8dc5-330aa291f487.png)

That image shows text boxes bleeding outside of the "white capsule" backgrounds that I put them in.  They are down and to the right in that example.

## Expected result

The Text Input and Text Label should be lined up in the center of the Sprite as per what the editor shows.

![image](https://user-images.githubusercontent.com/3253547/132071640-5534a05d-c0da-4fc0-aed2-32eac792f863.png)

Note the Blue background is a sprite behind the Input and Text objects lined up to surround them visually.

## More details

**Affected browsers/platforms:** 

Likely it's webkit browsers like Safari as Chromium based ones seem to be lined up nicely.

No specific Construct version.  Likely been like this for awhile with HTML controls over a Canvas.

Using current Stable release.

## System details

<details><summary>View details</summary>

Platform information
Browser: Edge
Browser version: 92.0.902.84
Browser engine: Chromium
Context: browser
Operating system: Windows
Operating system version: 10
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 16
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36 Edg/92.0.902.84
C3 release: r260 (stable)
Language setting: en-US

Local storage
Storage quota (approx): 140 gb
Storage usage (approx): 689 mb (0.5%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (NVIDIA)
Renderer: ANGLE (NVIDIA, NVIDIA GeForce RTX 3060 Ti Direct3D11 vs_5_0 ps_5_0, D3D11-27.21.14.6647)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by broknecho** [comment]
I think I found a way around the input box.  There is a CSS property called `box-sizing` that can be set to `border-box;` that seems to create a more reliable sizing for input boxes.

This seemed to work when adding it in the chrome dev tools: `box-sizing: border-box;`

Not sure you could add this as a default style on C3 inputs but may be a optional property checkbox in the `properties` panel that will enable/disable it so not to break backwards compatibility.

Still not sure on that Text Label as it's being rendered in the canvas.

**Comment by broknecho** [comment]
So Firefox shows it pretty good especially with the Text label:

![image](https://user-images.githubusercontent.com/3253547/132260273-0c8f0452-efdf-4848-a582-d60f8a5af602.png)

That is just from loading the URL at the beginning of the case. So likely both Webkit and Gecko seem to be affected by this.

**Comment by AshleyScirra** [comment]
I've added `box-sizing: border box` for form controls in the next release, which should get a more accurate size for the HTML elements that float over the canvas.

I'm afraid the issue with the Text object is unrelated and due to browsers having different interpretions of where the top baseline of text is. I don't think there's much we can do about it. I filed a Safari issue here: https://bugs.webkit.org/show_bug.cgi?id=226334
